### ISSUE TITLE
[BUG] <Technical Issue Report: Qwen Model Recommends Physically Incompatible Automotive Parts (Spark Plugs)>

### ISSUE THREAD

**Issue by Cheigo007** [QwenLM/Qwen#2263]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [x] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [x] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

Dear Alibaba Cloud Qwen Team,
 
I hope this email finds you well.
 
I am writing to report a systematic issue I encountered while using the Qwen model for technical automotive recommendations. I would greatly appreciate your feedback on this matter sent to my email address: [e-explor@mail.ru](mailto:e-explor@mail.ru)
 
 
Issue Summary
The Qwen model recommended spark plugs with incorrect physical specifications (thread size), making installation impossible. This represents a systematic error in prioritizing technical parameters.
 
 
Specific Case Details
Vehicle Context:
Car: Toyota. Daihatsu Terios J210, 2009, 4WD, Automatic Transmission
Engine: Daihatsu 3SZ-VE, 1.5L (1495 cc), DOHC 16V, VVT-i
Factory spark plugs: Denso K20PR-U11 (M12×1.25 thread, 19mm length, heat range 20, gap 1.1mm)
User installed: Denso XU22TT (M12×1.25 thread, heat range 22, gap 0.85mm)
 
User's Question: "Will Denso K20TT spark plugs work with my 3SZ-VE engine?"
 
Model's Response (Incorrect): The model recommended Denso K20TT as an "optimal choice," stating:
✅ Heat range 20 (matches factory)
✅ Gap 1.1mm (matches factory)
✅ Twin Tip technology (improved spark formation)
 
The Actual Problem: Denso K20TT spark plugs have M14×1.25 thread, while the 3SZ-VE engine requires M12×1.25 thread. Physical installation of K20TT in the 3SZ-VE cylinder head is impossible.
 
 
Root Cause Analysis
The model checked electrical/thermal parameters (heat range, gap) but ignored physical compatibility (thread size, thread length, socket size).
 
This is a systematic priority error:
 
Correct verification order should be:
Physical compatibility (thread, dimensions, fasteners) → BASIC LEVEL
Electrical parameters (heat range, resistance) → INTERMEDIATE LEVEL
Thermal mode and operating conditions → ADVANCED LEVEL
Optimization for usage pattern → EXPERT LEVEL
 
The model jumped to levels 2-3, skipping level 1.
 
 
Suggestions for Improvement
Option A: Prompt Engineering
Add to the system prompt for technical recommendations mandatory verification:
When recommending automotive parts, ALWAYS check in this order:
1. Physical compatibility (thread, dimensions, fasteners, connectors)
2. Electrical parameters (voltage, resistance, heat range)
3. Thermal mode and operating conditions
4. Compatibility with usage pattern
 
If unsure about physical parameters, honestly state:
"Need to verify in manufacturer catalog" instead of giving a recommendation.
 
Option B: Fine-tuning Dataset
Add to the training dataset examples of errors like:
"Recommending spark plugs with wrong thread size"
"Recommending filters with incorrect dimensions"
"Recommending brake pads for incompatible calipers"
 
With emphasis on prioritizing physical compatibility.
 
Option C: External Knowledge Base (RAG)
Integrate parts catalogs (TecDoc, manufacturer catalogs) to verify physical parameters before providing recommendations.
 
 
Impact Assessment
For Users:
Purchase of incompatible parts → financial loss (approximately $20-40 USD)
Time waste (returns, repeated searches)
Risk of damaging cylinder head threads during installation attempts
 
For Model Trust:
Users lose confidence in technical recommendations
Reduces the model's value as an expert tool
 
 
Additional Information
Date of conversation: 2026-06-08
Model version: Qwen 3.7-Plus
Query language: Russian
Full conversation log: Available upon request
 
 
Request for Feedback
I would be very grateful if you could:
 
Acknowledge receipt of this report
Provide feedback on whether this issue will be addressed in future model updates
Send your response to my email: [e-explor@mail.ru](mailto:e-explor@mail.ru)
 
I am also willing to:
Provide the complete conversation log
Answer any clarifying questions
Participate in testing fixes if needed
 
 
Closing
Thank you for your work on the Qwen model. This report is not criticism, but a contribution to improving the system. Users like me rely on Qwen for technical advice, and ensuring accuracy in safety-critical recommendations is essential.
 
I look forward to your response.
 
Best regards,
 
Igor,
Email: [e-explor@mail.ru](mailto:e-explor@mail.ru)
 
 
RUSSIAN VERSION
 
 
Тема: Отчёт о технической ошибке: модель Qwen рекомендует физически несовместимые автозапчасти (свечи зажигания)
 
Уважаемая команда поддержки Alibaba Cloud и разработчики модели Qwen,
Я пишу, чтобы сообщить о системной ошибке, с которой столкнулся при использовании модели Qwen для технических автомобильных рекомендаций. Буду очень признателен за обратную связь по этому вопросу на мой email адрес: [e-explor@mail.ru](mailto:e-explor@mail.ru)
 
 
Краткое описание проблемы
Модель Qwen рекомендовала свечи зажигания с неправильными физическими параметрами (размер резьбы), что делает установку невозможной. Это представляет собой системную ошибку в приоритезации технических параметров.
 
 
Детали конкретного кейса
Контекст автомобиля:
Автомобиль: Toyota. Daihatsu Terios J210, 2009, 4WD, АКПП
Двигатель: Daihatsu 3SZ-VE, 1.5L (1495 см³), DOHC 16V, VVT-i
Заводские свечи: Denso K20PR-U11 (резьба M12×1.25, длина 19 мм, калильное число 20, зазор 1.1 мм)
Установлено пользователем: Denso XU22TT (резьба M12×1.25, калильное число 22, зазор 0.85 мм)
 
Вопрос пользователя: "Подойдут ли свечи Denso K20TT для моего двигателя 3SZ-VE?"
 
Ответ модели (неправильный): Модель рекомендовала Denso K20TT как "оптимальный выбор", указав:
✅ Калильное число 20 (совпадает с заводским)
✅ Зазор 1.1 мм (совпадает с заводским)
✅ Технология Twin Tip (улучшенное искрообразование)
 
Фактическая проблема: Свечи Denso K20TT имеют резьбу M14×1.25, в то время как двигатель 3SZ-VE требует свечи с резьбой M12×1.25. Физическая установка K20TT в ГБЦ двигателя 3SZ-VE невозможна.
 
 
Анализ корня проблемы
Модель проверила электрические/тепловые параметры (калильное число, зазор), но проигнорировала физическую совместимость (размер резьбы, длина резьбы, размер ключа).
 
Это системная ошибка приоритетов:
 
Правильный порядок проверки должен быть:
Физическая совместимость (резьба, размеры, крепёж) → БАЗОВЫЙ УРОВЕНЬ
Электрические параметры (калильное число, сопротивление) → СРЕДНИЙ УРОВЕНЬ
Тепловой режим и условия эксплуатации → ПРОДВИНУТЫЙ УРОВЕНЬ
Оптимизация под режим использования → ЭКСПЕРТНЫЙ УРОВЕНЬ
 
Модель перепрыгнула на уровни 2-3, пропустив уровень 1.
 
 
Предложения по улучшению
Вариант А: Промпт-инжиниринг
Добавить в системный промпт для технических рекомендаций обязательную проверку:
 
При рекомендации автомобильных запчастей ВСЕГДА проверяй в следующем порядке:
1. Физическая совместимость (резьба, размеры, крепёж, разъёмы)
2. Электрические параметры (напряжение, сопротивление, калильное число)
3. Тепловой режим и условия эксплуатации
4. Совместимость с режимом использования
Если не уверен в физических параметрах — честно скажи:
"Нужно проверить в каталоге производителя", а не давай рекомендацию.
 
 
Вариант Б: Fine-tuning датасет
Добавить в обучающую выборку примеры ошибок типа:
"Рекомендация свечей с неправильной резьбой"
"Рекомендация фильтров с неправильными размерами"
"Рекомендация колодок для несовместимых суппортов"
 
С акцентом на приоритет физической совместимости.
 
Вариант В: Внешняя база знаний (RAG)
Интегрировать каталоги запчастей (TecDoc, каталоги производителей) для проверки физических параметров перед выдачей рекомендации.
 
 
Оценка влияния
Для пользователей:
Покупка неподходящих деталей → финансовые потери (приблизительно 1500-3000 руб.)
Потеря времени (возврат, повторный поиск)
Риск повреждения резьбы в ГБЦ при попытке установки
 
Для доверия к модели:
Пользователи теряют уверенность в технических рекомендациях
Снижается ценность модели как экспертного инструмента
 
 
Дополнительная информация
Дата диалога: 2026-06-08
Версия модели: Qwen 3.7-Plus
Язык запроса: Русский
Полный лог диалога: Доступен по запросу
 
 
Просьба об обратной связи
Буду очень признателен, если вы сможете:
 
Подтвердить получение этого отчёта
Предоставить обратную связь о том, будет ли эта проблема устранена в будущих обновлениях модели
Отправить ваш ответ на мой email: [e-explor@mail.ru](mailto:e-explor@mail.ru)
 
Также готов:
Предоставить полный лог диалога
Ответить на любые уточняющие вопросы
Участвовать в тестировании исправлений при необходимости
 
 
Заключение
Спасибо за вашу работу над моделью Qwen. Этот отчёт — не критика, а вклад в улучшение системы. Такие пользователи, как я, полагаются на Qwen для технических консультаций, и обеспечение точности рекомендаций, критичных для безопасности, крайне важно.
 
С нетерпением жду вашего ответа.
 
С уважением,
 
Игорь, [e-explor@mail.ru](mailto:e-explor@mail.ru)

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by Guciolek** [comment]
🤖 **Nabu auto-attempt** (auto-hunter v0.1)

Picked up by automated bounty scanner. Score: 88.0 | stack: 10 | competition: 0 comments
Fork prepared at Guciolek/Qwen — implementation in progress.

_Generated by `bounty-auto-hunter` skill_

**Comment by rehan243** [comment]
fwiw we hit something pretty similar in a medical device context — model recommends parts that *look* compatible, but it skips actual physical constraints unless you hard-bake them in. The thing is, Qwen (and honestly most LLMs) just match text patterns unless you force strict validation. For your spark plug thread/length mismatch, we ended up adding a post-processing filter step using a lookup dict with allowed combinations:

```python
allowed_parts = {
    "3SZ-VE": {
        "thread": "M12x1.25",
        "length": 19,
        "heat_range": 20
    }
}
if user_part["thread"] != allowed_parts["3SZ-VE"]["thread"] or user_part["length"] != allowed_parts["3SZ-VE"]["length"]:
    raise ValueError("Physically incompatible spark plug")
```

Not gonna lie, took us a few tries to get the lookup coverage, since docs for car parts are all over the place. We also had to retrain Qwen on a small synthetic dataset with physical failure cases to get it to flag these. If you’re running v1.8 or below, it trips up more on these edge cases.
